DNN Platform/DotNetNuke.Web/Common/DotNetNukeSecurity.cs

```diff
@@ -42,14 +46,19 @@ internal static class DotNetNukeSecurity
         private static FileSystemWatcher _fileWatcher;
         private static DateTime _lastRead;
         private static IEnumerable<string> _settingsRestrictExtensions = new string[] { };
+        private static Task _pendingTask;
+        private static IDictionary<int, bool> _taskStatus = new Dictionary<int, bool>();
 
         private const int CacheTimeOut = 5; //obtain the setting and do calculations once every 5 minutes at most, plus no need for locking
+        private const int InstallProcessingTime = 300; // do not start the watcher process in 5 minutes after database install complete.
+        private const int WaitingTime = 60; //let thread wait for 60s to check again.
+        private const int UpgradeIndicateWaitingTime = 180; //seconds to wait for indicate whether current changing is in upgrade process.
 
         // Source: Configuring Blocked File Extensions
         // https://msdn.microsoft.com/en-us/library/cc767397.aspx
         private static readonly IEnumerable<string> DefaultRestrictExtensions =
             (
-                ".ade,.adp,.app,.ashx,.asmx,.asp,.aspx,.bas,.bat,.chm,.class,.cmd,.com,.cpl,.crt,.dll,.exe," +
+                ".ade,.adp,.app,.ashx,.asmx,.asp,.aspx,.bas,.bat,.chm,.class,.cmd,.com,.cpl,.crt,.exe," +
```

Comment:
You MUST NOT remove the ``dll`` extension from this list - this can become a hole in site security.